GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+
+                string action = "";
+
+                switch (_state)
+                {
+                    case State.Bisect:
+                        action = "Bisect";
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Rebase:
+                        action = "Rebase";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Merge:
+                        action = "Merge";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        break;
+                    case State.Patch:
+                        action = "Patch";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                }
+
+                if (_state != State.None)
+                {
+                    string text = action + " currently in progress";
+                    if (_hasConflicts)
+                    {
+                        text += " with merge conflicts";
+                    }
+
+                    BackColor = _hasConflicts ? System.Drawing.Color.DarkOrange : System.Drawing.Color.LightSkyBlue;
+
+                    Label.Text = text;
+                    Show();
+                }
+                else
+                {
+                    Hide();
+                }
+            }
+        }
+
+        private void ResolveButton_Click(object sender, EventArgs e)
+        {
+            GitModuleForm form = FindForm() as GitModuleForm;
+            form.UICommands.StartResolveConflictsDialog(this);
+        }
+
+        private void ContinueButton_Click(object sender, EventArgs e)
+        {
+            GitModuleForm form = FindForm() as GitModuleForm;
+
+            switch (_state)
+            {
+                case State.Rebase:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ContinueRebaseCmd());
+                    break;
+                case State.Merge:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ContinueMergeCmd());
+                    break;
+                case State.Patch:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ResolvedCmd());
+                    break;
+            }
+
+            form.UICommands.RepoChangedNotifier.Notify();
+        }
+
+        private void AbortButton_Click(object sender, EventArgs e)
+        {
+            GitModuleForm form = FindForm() as GitModuleForm;
```

Comment:
I must be in front of my IDE to be sure but, except if `FormProcess.ShowDialog()` works well with a null value, I prefer using an explicit cast instead of a 'as'. It's easier to debug a cast exception than a NRE.

What do you think?

Reply:
Sure, this is my first time in C# so happy to learn 🙂

Reply:
ditto, as above